crypto/dsa/dsa.go

```diff
@@ -0,0 +1,87 @@
+package dsa
+
+import (
+	"fmt"
+
+	"github.com/tbd54566975/web5-go/crypto/dsa/ecdsa"
+	"github.com/tbd54566975/web5-go/crypto/dsa/eddsa"
+	"github.com/tbd54566975/web5-go/jwk"
+)
+
+var AlgorithmID = struct {
+	SECP256K1 string
+	ED25519   string
+}{
+	SECP256K1: ecdsa.SECP256K1AlgorithmID,
+	ED25519:   eddsa.ED25519AlgorithmID,
+}
+
+func GeneratePrivateKey(algorithmID string) (jwk.JWK, error) {
```

Comment:
The way these functions are repeated across all these packages indicates to me that they should be an interface. We can chat about how best to achieve that, though maybe there's something I'm missing.

Reply:
yeah would love to chat about that. I felt the same way, just didn't want to jump the gun until it seemed evident that a pattern was starting to form

Reply:
I think that's a good instinct to have, but this does seem like one of those moments where an interface would be useful.

Reply:
What you're implementing here looks like a registry pattern, which will benefit from the use of interfaces I think.